PROTO_tests/tests/bitops_test.py

```diff
@@ -0,0 +1,219 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+
+class TestBitOps:
+    @classmethod
+    def setup_class(cls):
+        cls.int_arr = ak.arange(10)
+        cls.uint_arr = ak.cast(cls.int_arr, ak.uint64)
+        cls.bigint_arr = ak.cast(cls.int_arr, ak.bigint)
+        cls.bigint_arr.max_bits = 64
+        cls.edge_cases = ak.array([-(2**63), -1, 2**63 - 1])
+        cls.edge_cases_uint = ak.cast(ak.array([-(2**63), -1, 2**63 - 1]), ak.uint64)
+        cls.edge_cases_bigint = ak.cast(cls.edge_cases_uint, ak.bigint)
+        cls.edge_cases_bigint.max_bits = 64
+
+    def test_popcount(self):
+        ans = [0, 1, 1, 2, 1, 2, 2, 3, 1, 2]
+        assert self.int_arr.popcount().to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert self.uint_arr.popcount().to_list() == ans_uint
+        assert self.bigint_arr.popcount().to_list() == ans_uint
+
+        # Edge case input
+        ans = [1, 64, 63]
+        assert ak.popcount(self.edge_cases).to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert ak.popcount(self.edge_cases_uint).to_list() == ans_uint
+        assert ak.popcount(self.edge_cases_bigint).to_list() == ans_uint
```

Comment:
so we're not gaining anything by converting the answer to `uint` here because it won't change the values. I should've known better when I originally added this. Once we realize that, we can clean this up a good deal

```python
for arr in self.int_arr, self.uint_arr, self.bigint_arr:
    assert arr.popcount().to_list() == [0, 1, 1, 2, 1, 2, 2, 3, 1, 2]

for arr in self.edge_cases, self.edge_cases_uint, self.edge_cases_bigint:
    assert ak.popcount(arr).to_list() == [1, 64, 63]
```
I think most of the functions can be updated in a similar way, so I won't comment on each one, but they should all be updated

Reply:
Updated